CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+# Contributing to awpy
+
+Hi! We're happy to have you here. Thank you in advance for your contribution to awpy.
+
+## Python code
+
+To make sure that awpy code stays readeable and works correctly we are making use of
+a variaty of helpfull tools.
+
+At the time of writing these are:
+
+- [black](https://github.com/psf/black): An uncompromising Python code formatter.
+- [ruff](https://github.com/astral-sh/ruff): An extremely fast Python linter, written in Rust.
+- [pyright](https://github.com/microsoft/pyright): A static type checker for Python.
+- [pylint](https://github.com/pylint-dev/pylint): A static code analyser for Python.
+- [pytest](https://docs.pytest.org/en/7.4.x/): A mature full-featured Python testing tool.
+- [pre-commit](https://pre-commit.com/): A framework for managing and maintaining multi-language pre-commit hooks.
+
+To make sure that you pull request is easy to review and can be merged quickly please
+please install these after cloning the repo.
+
+For that simply run the following commands:
+```shell
+pip install -r tests/requirements.txt
+pre-commit install --install-hooks
+```
+
+After you have made your changes locally run these tools to make sure there are no
+bugs or code smells have snook in. This can be done by running:
+
+```shell
+pre-commit run --all-files --show-diff-on-failure  # black formatting, ruff linting
+pyright  # type checking
+pylint awpy  # more linting
+coverage run -m pytest --durations=10  # python tests
+coverage report -m
+```
+
+The test part can take a while as it has to download a handful of demos to make sure
+that the parser is working correctly for all kinds of platforms and edge cases.
+
+If you do not have the time or bandwidth you can omit this part if absolutely necessary.
+The continous integration setup on GitHub will then run these for you.
+
+### Testing/Coverage
+
+If you are fixing a bug or adding a new feature make sure to also add [unit tests](https://en.wikipedia.org/wiki/Unit_testing)
+that cover the original bug or your new functionality. If you are new to that have a
```

Comment:
If you are new to writing unit tests, look at the aforementioned link, or check out the tests directory in awpy to see how our existing tests are built.